dandi/dandiapi.py

```diff
@@ -650,6 +650,23 @@ def __str__(self) -> str:
         return self.identifier
 
 
+class ValidationError(APIBase):
```

Comment:
```suggestion
class RemoteValidationError(APIBase):
    """
    .. versionadded:: 0.49.0

    Validation Error record obtained from a server. To not to be confused with
    :class:`dandi.validate_types.ValidationResult` which provides richer representation
    of validation errors.
```

as most of the constructs in this file have `Remote` in them. (note that sphinx ref to class might be wrong, typing from memory of how those are done)

But I also wonder if may be `ValidationResult` could just be used here instead?

Reply:
if switch to ValidationResult I would just keep all errors (for version and assets) in a single list but with a proper `scope` annotation within each ValidationResult.

Reply:
> if switch to ValidationResult I would just keep all errors (for version and assets) in a single list

That's not really up to us; the API returns the errors in two separate fields, and it wouldn't be worth the hassle to change that.

Reply:
Done: f1d752b

Reply:
well, we have those super-powers I would say, but ok -- let's stay close to the original return structure at this level.